reporters/ReportVectorMigration.cpp

```diff
@@ -125,6 +145,42 @@ GetReportInstantiator( Kernel::instantiator_function_t* pif )
             {
                 throw IncoherentConfigurationException( __FILE__, __LINE__, __FUNCTION__, "Start_Day", m_StartDay, "End_Day", m_EndDay );
             }
+            for (auto state : m_MustBeInState)
+            {
+                switch (state)
+                {
+                    case VectorStateEnum::STATE_EGG:
+                    {
+                        throw InvalidInputDataException(__FILE__, __LINE__, __FUNCTION__,
+                            "'STATE_EGG' is not a valid state for ReportVectorMigration.\n");
+                    }
+                    case VectorStateEnum::STATE_LARVA:
+                    {
+                        throw InvalidInputDataException(__FILE__, __LINE__, __FUNCTION__,
+                            "'STATE_LARVA' is not a valid state for ReportVectorMigration.\n");
+                    }
+                    case VectorStateEnum::STATE_IMMATURE:
+                    {
+                        throw InvalidInputDataException(__FILE__, __LINE__, __FUNCTION__,
+                            "'STATE_IMMATURE' is not a valid state for ReportVectorMigration.\n");
+                    }
```

Comment:
I think you could change this to something like:

                    case VectorStateEnum::STATE_EGG:
                    case VectorStateEnum::STATE_LARVA:
                    case VectorStateEnum::STATE_IMMATURE:
                    {
                        std::stringstream ss;
                        ss << "'" << VectorStateEnum::pairs::lookupKey( state ) << "'" is not a validate for ReportVectorMigration.";
                        throw InvalidInputDataException(__FILE__, __LINE__, __FUNCTION__, ss.msg().c_str );
                    }